codegen/lib/graphql_java_gen/templates/APISchema.java.erb

```diff
@@ -306,11 +310,15 @@ public class <%= schema_name %> {
                     <%= generate_build_input_code(escape_reserved_word(field.camelize_name), field.type) %>
                   <% end %>
                   <% type.optional_input_fields.each do |field| %>
-                    if (<%= escape_reserved_word(field.camelize_name) %> != null) {
+                    if (this.<%= escape_reserved_word(field.name) %>Seen) {
```

Comment:
Again, use `field.camelize_name` without `escape_reserved_word` to avoid errors if this does get escaped